crates/provider-sqldb-postgres/wit/deps/postgres/connection.wit

```diff
@@ -0,0 +1,50 @@
+package wasmcloud:postgres@0.1.0-draft;
+
+/// Manual connection management
+interface connection {
+  use types.{create-connection-error};
+
+  /// Options for creating a connection to a Postgres instance
+  record connection-create-options {
+    /// Host to connect to
+    host: string, 
+    /// Port
+    port: u16, 
+    /// Username
+    username: string, 
+    /// Password
+    password: string, 
+    /// Database
+    database: string, 
+    /// Require TLS for connections
+    tls-required: bool, 
+  }
+
+  /// To support both managed connections (determined by link configuration)
+  /// and component-driven manual connection management (i.e. this WIT interface)
+  /// we allow connection options to take multiple forms.
+  ///
+  /// By default, connections can be:
+  /// - managed completely by the provider, with the component dealing only in terms of queries
+  /// - component-driven with a named-profile selected by the component
+  /// - component-driven with connection details specified completely by the component
+  variant connection-options {
+    /// Default completely managed connection
+    managed,
+    /// An existing named profile
+    named-profile(string),
+    /// Near-completely dynamic connection creation
+    ///
+    /// NOTE: components probably won't need this, but more dynamic setups might, we can also disable this kind by default at the provider level
+    manual(connection-create-options),
+  }
+
+  /// A token that represents a connection of some sort,
+  /// either a raw database connection or a database session.
+  ///
+  /// This token can be expected to be somewhat opaque to users.
+  type connection-token = string;
```

Comment:
Why isn't this modeled as a `resource` if it is meant to be an opaque reference to something?

Reply:
Because IIRC resources aren't quite yet supported by the Rust tooling! Our Golang tooling actually leapfrogged our Rust and resources are supported but require some extra work right now -- one of the reasons this design is `0.1.0-draft`!

There are two things that we really should have that will come later:

- Resources (`resource connection`/`resource session`)
- Async (for `stream<result-row>`)

Reply:
Hey so note also that for the first version of this we're going to [simplify the interface *even* further](https://github.com/wasmCloud/wasmCloud/pull/2129#issuecomment-2138041188) -- we're going to remove the manual connection stuff as a possibility. 

In the future we *will* still want the `stream` stuff, but for this version you won't even need the pseudo-resource `connection-token`.

Reply:
I'm slightly tempted to implement a manual cursor interface as the return (rather than `list<result-row>`), but for now let's keep it simple and just wait for `stream<result-row>` to be possible

Reply:
I think keeping it simple with a list and aiming for a stream later is fine.
That said, I have the same question for the prepared statements and filed a separate comment for that.